lib/ansible/galaxy/collection.py

```diff
@@ -1119,6 +1298,32 @@ def _get_collection_info(dep_map, existing_collections, collection, requirement,
     dep_map[to_text(collection_info)] = collection_info
 
 
+def parse_scm(collection, version):
```

Comment:
Why didn't you use urllib split?

Reply:
Mostly for similarity with roles. If you have a recommendation for a better way I'd update this.

Reply:
I didn't know the detail about roles. It just looked like a lot of manual processing which may be error-prone.